internal/jobs/launch_instance_aws.go

```diff
@@ -43,6 +46,18 @@ func HandleLaunchInstanceAWS(ctx context.Context, job dejq.Job) error {
 		return fmt.Errorf("unable to decode args: %w", err)
 	}
 
+	jobErr := handleLaunchInstanceAWS(ctx, &args)
+	if jobErr != nil {
+		finishWithError(ctx, args.ReservationID, jobErr)
+	}
+	return jobErr
+}
+
+// Job logic, when error is returned the job status is updated accordingly
+func handleLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
```

Comment:
I am splitting handlers into exported and unexported function, the former will unmarshal and handle error. This is common across all the three jobs we have today. We kind of ignored errors until now, my bad.